lib/src/chart/base/axis_chart/axis_chart_painter.dart

```diff
@@ -214,11 +214,9 @@ abstract class AxisChartPainter<D extends AxisChartData>
       final from = Offset(0, getPixelY(line.y, viewSize, holder));
       final to = Offset(viewSize.width, getPixelY(line.y, viewSize, holder));
 
-      final isLineBeingDrawnOutsideChart = from.dy <= 0 || to.dy <= 0;
-      final isLineBeingDrawnOnXAxis =
-          from.dy == viewSize.height || to.dy == viewSize.height;
+      final isLineBeingDrawnOutsideChart = from.dy < 0 || to.dy < 0;
 
-      if (!(isLineBeingDrawnOutsideChart || isLineBeingDrawnOnXAxis)) {
+      if (!isLineBeingDrawnOutsideChart) {
```

Comment:
What if we remove the whole if statement? Because I don't see any if statement in the `drawVerticalLines` function.
BTW if you want to keep the restriction for drawing lines inside the chart, I think you need to check both bottom and top bounds:

```dart
final isLineBeingDrawnOutsideChart = from.dy < 0 || to.dy < 0 || from.dy > viewSize.height || to.dy > viewSize.height;
...
```

Also please update the `drawVerticalLines` function to restrict the drawing between `0` and `viewSize.width`

Reply:
I'd be okay removing the whole if statement. It was my attempt at preventing lines being drawn off the chart. But...maybe we should just leave that up to the developers to ensure they aren't drawing any lines off the chart instead? If you agree I'll remove this check and won't make any updates to `drawVerticalLines`.

Reply:
In my opinion, it is better to restrict it. Because there are some issues that developers complained about why the line draws out of the chart.
Can we check it with the chart's `minX`, `maxX`, `minY`, and `maxY`?

Reply:
Sounds good. I'll look into restricting drawing of extra lines inside of the bounds. I'm about to head on some PTO so I will not be around for about a week.

Reply:
Great!
Take your time.

Reply:
> Great! Take your time.

Just pushed up some changes. Extra lines are now being drawn on x and y min/max values and are ignored if they are less than min or greater than max.